sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/PartitionKeyBuilder.java

```diff
@@ -118,4 +122,37 @@ public PartitionKey build() {
         partitionKeyInternal = PartitionKeyInternal.fromObjectArray(valueArray, true);
         return new PartitionKey(partitionKeyInternal);
     }
+
+    /**
+     * Returns the PartitionKey from an array of objects that is generated by PartitionKeyInternal
+     * @param values The object array of values to construct the PartitionKey with
+     * @param strict The boolean value for if to use strict on object types
+     * @return The PartitionKey
+     */
+    public static PartitionKey fromObjectArray(Object[] values, boolean strict) {
+        if (values == null) {
+            throw new IllegalArgumentException("values can't be null");
+        }
+        return new PartitionKey(PartitionKeyInternal.fromObjectArray(values, strict));
+    }
+
+    /**
+     * Returns the PartitionKey extracted from the document by PartitionKeyHelper
+     * @param document The JsonSerializable object to get the PartitionKey value from
+     * @param partitionKeyDefinition The PartitionKeyDefinition to use to extract the PartitionKey value
+     * @return The PartitionKey
+     */
+    public static PartitionKey extractPartitionKeyFromDocument(
+        Map<String, Object> document,
+        PartitionKeyDefinition partitionKeyDefinition) throws JsonProcessingException {
+        if (document == null) {
+            throw new IllegalArgumentException("document can't be null");
+        }
+        if (partitionKeyDefinition == null) {
+            throw new IllegalArgumentException("partitionKeyDefinition can't be null");
+        }
+        ObjectMapper objectMapper = new ObjectMapper();
+        return PartitionKeyHelper.extractPartitionKeyFromDocument(
+            new JsonSerializable(objectMapper.writeValueAsString(document)), partitionKeyDefinition);
```

Comment:
no need to convert into JsonSerializable.

convertValue to ObjectNode should be sufficient

Reply:
I tried that initially but it doesn't like the conversion
![image](https://github.com/Azure/azure-sdk-for-java/assets/63077034/fbff60f8-7d52-4b76-8585-7f38bac5c3e0)